README.md

```diff
@@ -1,24 +1,241 @@
+Turbolinks. Turbolinks is a JavaScript library that makes your application
+behave as if it were a single-paged application.
 
-* Ruby version
+## Getting started
 
-* System dependencies
+> __Important:__ The session token feature is currently restricted to 
+> participants of a private beta. The feature is scheduled for general release in
+> Summer 2020.
 
-* Configuration
+If you are looking to install this app on your own testing environment, and
+already have a `Shopify API Key` and `Shopify API Secret` [from your partners
+dashboard][7], you can pull this repository and run the following:
 
-* Database creation
+```console
+$ bundle install
+$ yarn install
+$ rails db:migrate
+```
+See [this section][3] for a guide on setting up your `.env` file.
 
-* Database initialization
+You will also need to use [ngrok][4] to make `localhost:3000` available to the
+internet. You should use the secure HTTPS URL generated by ngrok to create your
+app with on your Partners dashboard.
 
-* How to run the test suite
+To start the app, simply run:
 
-* Services (job queues, cache servers, search engines, etc.)
+```console
+$ rails s
+```
 
-* Deployment instructions
+Your requests to protected resources should be secured with an `Authorization:
+Bearer token` header.
 
-* ...
+![App dashboard][s2]
+
+_**Above:** A sample multi-paged app with an authenticated landing page._
+
+![App requests][s1]
+
+_**Above:** Requests made to protected resources are authenticated using JWTs._
+
+## Enabling Turbolinks on your app
+
+The official [Turbolinks GitHub guide][2] is an excellent resource on getting
+started with Turbolinks. Below are the steps taken to bootstrap Turbolinks on
+this sample app.
+
+1. Add the `turbolinks` gem to your Gemfile:
+
+```rb
+gem 'turbolinks', '~> 5'
+```
+
+2. Run `bundle install`.
+
+3. Add the `turbolinks` package to your application: `yarn add turbolinks`.
+
+4. Add the following line to `app/javascript/packs/application.js` if your app
+uses webpack to manage its manifest files:
+
+```js
+require("turbolinks").start()
+```
+
+## Making authorized requests to protected resources
+
+Before you can navigate to protected resources using Turbolinks, you should
+fetch an authenticated session token. This sample app creates a `/widgets`
+resource that is protected. It requires JWT authorization to read and
+write.
+
+1. In `config/initializrs/shopify_app.rb` add the following lines to the `ShopifyApp.configure` block:
+
+```rb
+ShopifyApp.configure do |config|
+  …
+  config.allow_jwt_authentication = true
+  config.myshopify_domain = ENV['SHOPIFY_DOMAIN']
+end
+
+ShopifyAPI::Session.myshopify_domain = ENV['SHOPIFY_DOMAIN']
+```
+
+* `config.allow_jwt_authentication = true` ensures that your app is configured to allow JWT-based
+  authentication
+* `config.myshopify_domain` ensures that the install link for the app goes to the correct URL
+* `ShopifyAPI::Session.myshopify_domain` ensures that the app sends requests to
```

Comment:
I don't think this is useful in a public readme. Since partners will always be using production Shopify to test. Maybe we can delete this.